oak/server/oak_node.h

```diff
@@ -82,6 +80,20 @@ class OakNode final : public Node::Service {
   const std::string module_hash_sha_256_;
 };
 
+// RAII class to add a mapping from a handle to a channel half for the duration of a scope.
```

Comment:
Nice pattern!

Reply:
http://go/totw/41 ...

Reply:
You may like Rust hash map [entry API](https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html) then